Scripts/demand/trips.py

```diff
@@ -86,24 +90,35 @@ def create_population(self):
         list
             Person
         """
-        self.cm.calc_basic_prob()
         self.population = []
         zones = self.zone_data.zone_numbers[:self.zone_data.first_peripheral_zone]
-        for idx in zones:
+        for zone_number in zones:
             weights = [1]
             for age_group in self.age_groups:
                 key = "share_age_" + str(age_group[0]) + "-" + str(age_group[1])
-                share = self.zone_data[key][idx]
+                share = self.zone_data[key][zone_number]
                 weights.append(share)
                 weights[0] -= share
-            for _ in xrange(0, int(self.zone_data["population"][idx])):
-                a = numpy.arange(-1, len(self.age_groups))
-                group = numpy.random.choice(a=a, p=weights)
-                if group != -1:
-                    # Group -1 is under-7-year-olds and they have weights[0]
-                    age_group = self.age_groups[group]
-                    person = Person(idx, age_group, self.gm, self.cm)
-                    self.population.append(person)
+            weights[0] = max(weights[0], 0)
+            if sum(weights) > 1:
+                if sum(weights) > 1.005:
+                    msg = "Sum of age group shares for zone {} is {}".format(
+                        zone_number, sum(weights))
+                    log.error(msg)
+                    raise ValueError(msg)
+                else:
+                    weights = numpy.array(weights)
+                    rebalance = 1 / sum(weights)
+                    weights = rebalance * weights
+            for _ in xrange(0, int(self.zone_data["population"][zone_number])):
+                if random.random() < param.agent_demand_fraction:
```

Comment:
This is a good idea. I think we should then also `round()` the population, because otherwise `int()` always rounds down.

> `int(round(self.zone_data["population"][zone_number]*param.agent_demand_fraction))`